Prepare label screen directives for dynamic surveys + notes

JGreenlee opened the pull request:
<img width="458" alt="image" src="https://user-images.githubusercontent.com/15843932/208820626-c6306712-c1b7-4d1b-b22e-6b3d4643fc2b.png">

temporarily hardcoded `showPlaces = true` in `infinite_scroll_list.js`, as a placeholder for the property we will be adding to the config

JGreenlee committed:
create 'place list item' directive

JGreenlee committed:
style the timeuse/activity fab

JGreenlee committed:
prevent box shadows of diary-items being cut off

JGreenlee commented:
<img width="590" alt="image" src="https://user-images.githubusercontent.com/15843932/208821162-679fa8a4-3159-4234-a11c-dc12b94dda26.png">

@shankari Do these timestamps need to be added on the place list item? They seem redundant in this mockup example - but are there real-life scenarios where they won't overlap?

shankari commented:
@JGreenlee they should overlap in all real-world scenarios. @asiripanich FYI

asiripanich commented:
> @JGreenlee they should overlap in all real-world scenarios. @asiripanich FYI

I agree that they should overlap in all real-world scenarios. However, we might want to think about the case when there is a 'not valid' trip on the screen.

shankari commented:
@asiripanich can you expand further? why do we need to do anything specific for that case?
@JGreenlee although you don't display it, you should make sure that you have the start and end times accessible in the directive so that we can pre-populate the start and end times of the popped up survey with the start and end times of the place item.

shankari commented on www/js/diary/infinite_scroll_place_item.js:
```diff
@@ -0,0 +1,39 @@
+  .controller("PlaceItemCtrl", function($scope, $state) {
+    console.log("Place Item Controller called");
+
+    // TODO detail page for places
+    // $scope.showDetail = function($event) {
+    //   $state.go("root.main.inf_scroll-detail", {
+    //  
+    //   });
+    // }
```
I don't think we need a detail page for places; the plan right now is to have a list of place based user inputs in the item. Note that this implies that the item needs to resize when we add new place based user inputs.

JGreenlee replied:
I was going to ask about whether a detail page is necessary. Thanks for clearing that up 👍

`infinite_scroll_list` uses `collection-repeat`, which relies on each item having a fixed height. To have dynamic heights, we will need a different scrolling solution.

I can easily just swap in `ng-repeat`, but I believe scrolling will be less performant (if we have hundreds of trips). Best solution would be to tap into native scrolling. But can we even do that in this version of ionic?

shankari replied:
@JGreenlee I came up with a solution to resize collection-repeat elements - we needed it for the replaced mode display. Please check with @sebastianbarry on details.

shankari commented on www/templates/diary/infinite_scroll_list.html:
```diff
@@ -41,6 +41,7 @@ <h2 class="hr-lines" translate> {{trip.display_date}}</h2>
                 this and implement some sort of horizontal line between trips-day-changing later on-->
 
                 <infinite-scroll-trip-item trip="trip" map-limiter="mapLimiter"></infinite-scroll-trip-item>
+                <infinite-scroll-place-item ng-if="showPlaces" trip="trip"></infinite-scroll-trip-item>
```
pro/con for having the `ng-if` here versus in the `place_list_item.html`. I have a mild preference to keep the logic in the directive instead of the list. In either case, I would like to document the rationale behind the decision for the future.

JGreenlee replied:
Up to you. I prefer to hide/show elements at their root because I don't want empty containers in the DOM

I can move the `ng-if` to `place_list_item.html`, but when the config is set `false` I believe we will have an empty `<infinite-scroll-place-item>`.

shankari replied:
I'm fine with leaving it here, but can you please add a comment with this rationale, so we don't have somebody looking at this 5 years from now saying why? why?

shankari commented on www/templates/diary/trip_list_item.html:
```diff
@@ -1,7 +1,7 @@
 <!-- New Trip Template format  -->
 
-    <ion-item id="unified-diary-item" style="background-color: transparent;" class="list-item">
+    <ion-item class="unified-diary-item" style="background-color: transparent;" class="list-item">
```
Yes! I have no idea why we used `id` instead of `class` in the first place.

shankari commented on www/templates/survey/enketo/timeuse-button.html:
```diff
@@ -1,6 +1,5 @@
-    ng-click="openPopover($event)"
-  ></button>
+  <button class="button icon ion-plus button-primary fab-button-shadow" ng-click="openPopover($event)">
+    <b>Add Activity</b>
```
This should be, at a minimum, internationalized. Please see the way that other text in the app is represented.
Ideally, we would also have this text come in as part of the config so that we can support other use cases like the "missing trip"

JGreenlee replied:
I was wondering about that. will ask today

shankari reviewed (commented):
This is a great start - I have some minor comments. After these are resolved, we should discuss the storage/retrieval as well so you can actually start displaying the values.

JGreenlee committed:
Merge branch 'places-survey' into places-survey

sebastianbarry committed:
Removed enketoSurveyConfig (replaced in deploy_configs)

service.js
- Added DynamicConfig library
- Removed the ENKETO_SURVEY_CONFIG_PATH since we are getting the path from the dynamic config now
- Added the DynamicConfig get text which gets the config.surveys data to be used
- Removed the _lazyLoadConfig http request now that we are getting the survey json path from the dynamic config data

answer.js
- Added DynamicConfig library
- Removed the ENKETO_SURVEY_CONFIG_PATH since we are getting the path from the dynamic config now
- Added the DynamicConfig get text which gets the config.surveys data to be used
- Removed the _lazyLoadConfig http request now that we are getting the survey json path from the dynamic config data

enketo-time-use.js
- Fixed the location of the formPath file since it was changed in nrel_openpath_deploy_configs

Removed enketoSurveyConfig.json.sample because we no longer need it due to the data being in nrel_openpath_deploy_configs

asiripanich commented:
> @asiripanich can you expand further? why do we need to do anything specific for that case?

Think of the case where the user had activities *during* or *after* the duration of an invalid trip. 
- In the case of *during*, activities would be added to the invalid trip. 
- In the case of *after*, the location displayed in the between trips activity card would be incorrect.

![image](https://user-images.githubusercontent.com/17020181/209244795-1218c376-8401-42b6-a16e-ff8f3a44c503.png)

shankari commented:
@asiripanich is your expectation that invalid trips will not be displayed? Because that is not the current behavior and has never been the behavior to my knowledge.

Also, in general, invalid trips are round trips - they seem jump to another location and then come back - so not sure why the location between the cards would be incorrect. Have you seen other types of invalid trips? Can you give us some examples?

JGreenlee committed:
generalize 'time use' to 'notes'

JGreenlee committed:
adjust label screen directives to dynamic config

JGreenlee committed:
Merge branch 'places-survey' of https://github.com/sebastianbarry/e-mission-phone into places-survey

JGreenlee commented:
Added my progress to this PR
<img width="492" alt="image" src="https://user-images.githubusercontent.com/15843932/209309251-20466be7-d6f0-488f-bf9d-607a92eafae1.png">

Temporarily assuming the config structure I suggested [here](https://github.com/e-mission/nrel-openpath-deploy-configs/pull/7#issuecomment-1363340694). If we go a different route I can adjust this accordingly.

My next task, I think, is to create a small list to display notes under the add button. Once we can handle results from the surveys and store them, I'll be able to propagate the list with 'activity' or 'missed trips'.

asiripanich commented:
> @asiripanich is your expectation that invalid trips will not be displayed? Because that is not the current behavior and has never been the behavior to my knowledge.

Ideally, it should not interfere with time use activity logging. Should invalid trips be displayed? Yes, but maybe make them less visible? 

> Also, in general, invalid trips are round trips - they seem jump to another location and then come back - so not sure why the location between the cards would be incorrect. Have you seen other types of invalid trips? Can you give us some examples?

It is true that they are mostly round trips, hence, like you mentioned, location is not the problem. The problem is that any activities happened during the duration of an invalid trip would need to be logged as activities 'during' the invalid trip.

JGreenlee committed:
Merge branch 'places-survey' of https://github.com/sebastianbarry/e-mission-phone into places-survey

JGreenlee committed:
refactor so that dyn. config is only accessed once for label screen directives

JGreenlee committed:
apply surveyOpt based on dynamic config

JGreenlee committed:
tweak enketo button style
